src/Evolution/Initialization/AddToDataBox.hpp

```diff
@@ -0,0 +1,117 @@
+                << pretty_type::get_name<AddingAction>()
+                << " is not the same as what is already in the DataBox. The "
+                   "value in the DataBox is: "
+                << db::get<tag>(box) << " while the value being added is "
+                << tuples::get<tag>(simple_tags));
+        }
+        return nullptr;
+      },
+      [](auto tag_v, auto overwrite_simple_vars)
+          -> Requires<not decltype(overwrite_simple_vars)::value and
+                      not tt::has_inequivalence_v<
+                          db::item_type<typename decltype(tag_v)::type>>> {
+        using tag = typename decltype(tag_v)::type;
+        Parallel::printf(
+            "Note: Unable to check that the initial value of tag '%s' being "
+            "re-added is the same as that in the DataBox. The action re-adding "
+            "the tag is '%s'\n.",
+            pretty_type::get_name<tag>(),
+            pretty_type::get_name<AddingAction>());
+        return nullptr;
+      },
+      [&simple_tags, &box](auto tag_v, auto overwrite_simple_vars)
+          -> Requires<decltype(overwrite_simple_vars)::value> {
+        using tag = typename decltype(tag_v)::type;
+        db::mutate<tag>(
+            make_not_null(&box), [&simple_tags](const auto item_ptr) noexcept {
+              *item_ptr = tuples::get<tag>(simple_tags);
+            });
+        return nullptr;
+      });
+  (void)check_value_helper;  // avoid compiler warnings
+  EXPAND_PACK_LEFT_TO_RIGHT(
+      check_value_helper(tmpl::type_<SimpleTagsToCheck>{},
+                         std::integral_constant<bool, OverwriteSimpleVars>{}));
+
+  // Only add compute tags that are not in the DataBox.
+  using compute_tags_to_add = tmpl::remove_if<
+      ComputeTagsList,
+      tmpl::bind<tmpl::list_contains,
+                 tmpl::pin<typename db::DataBox<DbTagsList>::compute_item_tags>,
+                 tmpl::_1>>;
+
+  return db::create_from<db::RemoveTags<>,
+                         db::AddSimpleTags<SimpleTagsToAdd...>,
+                         compute_tags_to_add>(
+      std::move(box), std::move(tuples::get<SimpleTagsToAdd>(simple_tags))...);
+}
+}  // namespace detail
+
+/*!
+ * Adds tags that are not yet in the DataBox.
+ *
+ * If `OverwriteSimpleVars` is `true` then the values of simple tags already in
+ * the DataBox will be overwritten by the call to this function. If
+ * `OverwriteSimpleVars` is `false` then for simple tags whose type have an
+ * inequivalence operator a check will be done that they have the same value,
+ * for ones without an inequivalence operator a note will be printed to screen
+ * to inform users of a potential issue. Compute tags that are not in the
+ * DataBox are added, ones that are in the DataBox already are ignored.
+ */
+template <typename AddingAction, typename ComputeTagsList,
+          bool OverwriteSimpleVars = false, typename... SimpleTags,
+          typename DbTagsList>
+auto add_nonexistent_to_databox(
```

Comment:
I don't think it should be used outside of initialization. I'm worried it'll get people into the bad habit of rather than understanding and fixing a compiler error, just saying "Meh, ignore duplicates". I'm really not open to moving this for that reason. I thought about it a lot while writing it and I think it's a bad decision.

Reply:
I'm not sure, need to think about it more :)

Reply:
After more thought and discussion I agree with this.